### ISSUE TITLE
Cannot purge some tasks

### ISSUE THREAD

**Issue by gms8994** [GothenburgBitFactory/taskwarrior#2254]
Running 2.6.0 (4b67f10cb), I cannot purge some tasks by uuid. I'm attempting to purge all tasks from a specific project with the following (there's probably a more elegant way of doing this, but scripting is my strong suit):

```
task export project:MyProject | jq -r '.[].uuid' | xargs -I{} task rc.confirmation=off purge {} 
```

Many of these show the following output:

```
Filter: ( project = MyProject)
Purged 0 tasks.
Configuration override rc.confirmation=off
Filter: ( ( ( uuid = e5b03d47-d55a-4d62-aa40-44e9061eeed9 ) ) )
```

`task diag` output:

```
task 2.6.0
   Platform: Darwin

Compiler
    Version: 4.2.1 Compatible Apple LLVM 11.0.3 (clang-1103.0.32.29)
       Caps: +stdc +stdc_hosted +LP64 +c8 +i32 +l64 +vp64 +time_t64
 Compliance: C++14

Build Features
      Built: Apr 13 2020 16:36:41
     Commit: 4b67f10cb
      CMake: 3.17.0
    libuuid: libuuid + uuid_unparse_lower
  libgnutls: 3.6.12
 Build type: release

Configuration
       File: /Users/gsolsberry/.taskrc (found), 4913 bytes, mode 100644
       Data: /Users/gsolsberry/.task (found), dir, mode 40755
    Locking: Enabled
         GC: Enabled
    $EDITOR: vim
     Server: freyr.expansesoftware.com:53589
         CA: /Users/gsolsberry/.task/ca.cert.pem, readable, 3751 bytes
Certificate: /Users/gsolsberry/.task/private.cert, readable, 3735 bytes
        Key: /Users/gsolsberry/.task/private.key, readable, 24737 bytes
      Trust: ignore hostname
    Ciphers: NORMAL
      Creds: glens/Glen Solsberry/************************************

Hooks
     System: Enabled
   Location: /Users/gsolsberry/.task/hooks
             (-none-)

Tests
   Terminal: 120x30
       Dups: Scanned 3979 tasks for duplicate UUIDs:
             No duplicates found
 Broken ref: Scanned 3979 tasks for broken references:
             No broken references found
```

Attempting to manually purge the tasks by uuid also fails, though it does look like task finds the task (as seen by `Filtered 3979 tasks --> 1 tasks [all tasks]`).

```
t rc.debug.parser=1 e5b03d47-d55a-4d62-aa40-44e9061eeed9 purge
Purged 0 tasks.
Timer Config::load (/Users/gsolsberry/.taskrc) 0.001179 sec
No context.
Parse Tree (before command-specifіc processing)
  _original_args
    task rc.debug.parser=1 e5b03d47-d55a-4d62-aa40-44e9061eeed9 purge
  _args
    word basename='task' raw='task' BINARY 
    pair modifier='debug.parser' name='rc' raw='rc.debug.parser=1' separator='=' value='1' CONFIG ORIGINAL 
    op raw='(' ORIGINAL FILTER 
    uuid raw='e5b03d47-d55a-4d62-aa40-44e9061eeed9' ORIGINAL FILTER 
    op raw=')' ORIGINAL FILTER 
    identifier canonical='purge' raw='purge' ORIGINAL CMD RUNSGC USESCONTEXT ALLOWSFILTER 

Filtered 3979 tasks --> 1 tasks [all tasks]
  pending.data rw - T0015+000~000-000 L0015+000
completed.data rw - T3964+000~000-000 L3964+000
     undo.data rw - T0000+000~000-000 L0000+000
  backlog.data rw - T0000+000~000-000 L0000+000
 
Perf task 2.6.0 4b67f10cb 20200413T210828Z init:4614 load:253157 gc:36 filter:194313 commit:551 sort:0 render:0 hooks:3 other:30296 total:482970

Configuration override rc.debug.parser=1
Filter: ( ( ( uuid = e5b03d47-d55a-4d62-aa40-44e9061eeed9 ) ) )
```

Running `task info` correctly finds and displays the task:

```
t rc.debug.parser=1 e5b03d47-d55a-4d62-aa40-44e9061eeed9 info
Timer Config::load (/Users/gsolsberry/.taskrc) 0.001245 sec
Parse Tree (before command-specifіc processing)
  _original_args
    task rc.debug.parser=1 e5b03d47-d55a-4d62-aa40-44e9061eeed9 info
  _args
    word basename='task' raw='task' BINARY 
    pair modifier='debug.parser' name='rc' raw='rc.debug.parser=1' separator='=' value='1' CONFIG ORIGINAL 
    op raw='(' ORIGINAL FILTER 
    uuid raw='e5b03d47-d55a-4d62-aa40-44e9061eeed9' ORIGINAL FILTER 
    op raw=')' ORIGINAL FILTER 
    identifier canonical='information' raw='info' ORIGINAL CMD READONLY ALLOWSFILTER 

Filtered 3979 tasks --> 1 tasks [all tasks]
  pending.data rw - T0015+000~000-000 L0015+000
completed.data rw - T3964+000~000-000 L3964+000
     undo.data rw - T0000+000~000-000 L14273+000
  backlog.data rw - T0000+000~000-000 L0000+000
 
Perf task 2.6.0 4b67f10cb 20200413T211209Z init:8065 load:213681 gc:0 filter:397165 commit:714 sort:0 render:0 hooks:7 other:125831 total:531782


Name          Value                                                                                                         
ID            -                                                                                                             
Description   Staging Data                                                                                                  
Status        Completed                                                                                                     
Project       MyProject
Entered       2019-02-13 07:15:44 (1.2y)                                                                                    
Due           2019-02-13 17:00:00
End           2019-02-13 14:30:27                                                                                           
Last modified 2019-02-13 14:30:27 (1.2y)
Tags          abn                                                                                                           
Virtual tags  COMPLETED ORPHAN TAGGED UNBLOCKED PROJECT
UUID          e5b03d47-d55a-4d62-aa40-44e9061eeed9                                                                          
Urgency       15.8
[estimate     0.500000]                                                                                                     

    project      1 *    1 =      1
    tags       0.8 *    1 =    0.8
    due          1 *   12 =     12
    age          1 *    2 =      2
                            ------
                              15.8

Configuration override rc.debug.parser=1
Filter: ( ( ( uuid = e5b03d47-d55a-4d62-aa40-44e9061eeed9 ) ) )
```

**Comment by gms8994** [comment]
Looks as though tasks must be in the `status:deleted` state before `task purge` will work. That documentation exists in the man page, but that wasn't noticed by me on the first pass (it's not noted on the website). Should the website documentation be updated to reflect that?

**Comment by tbabej** [comment]
Fixed in 2.6.0 via #2358.
